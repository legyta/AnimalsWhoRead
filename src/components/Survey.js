import React from "react";
import { Link } from "react-router-dom";
import "./Survey.scss";
import frame from "./frame.svg";

const Survey = props => (
  <section id="survey">
    <img src={frame} alt="frame" classApp="App-frame" />
    <div className="title-description">
      <div className="title">
        <h1>Feelings about Extinction</h1>
      </div>
      <div className="description">
        <p>Greetings to animals sharing their feelings here</p>
      </div>
    </div>
    <div className="contact">
      <div className="name">
        <label for="name" id="name-label">
          How are you called
        </label>

        <input
          type="text"
          id="name"
          name="name"
          placeholder="your species name"
          required
        />
      </div>
      {/* email  */}
      <div className="email">
        <label for="email" id="email-label">
          What is your email
        </label>

        <input
          type="email"
          id="email-input"
          name="email"
          placeholder="if you have one"
          pattern="[^ @]*@[^ @]*"
        />
      </div>
      {/* age */}
      <div className="age">
        <label for="number" id="number-label">
          Years on the planet
        </label>

        <input
          type="number"
          id="number"
          name="number"
          placeholder="age"
          min="0"
          max="100"
        />
      </div>
    </div>
    <div className="dropdown-list">
      <label id="dropdown-title">How do you feel about Extinction?</label>
      <select id="dropdown">
        <option value="current">Choose what you relate most to</option>
        <option value="first-choice">
          I am very afraid! I already store food and think of new possible
          shelter.
        </option>
        <option value="second-choice">We have a plan to fight it!</option>
        <option value="third-choice">
          I am scared, but I don't care. My elders also got extincted.
        </option>
        <option value="fourth-choice">
          Huh? What is that? Did I sleep for too long again. Need to catch up on
          the news!
        </option>
        <option value="fifth-choice">Prefer not to answer.</option>
      </select>
    </div>
    <div className="radio-list">
      <label id="radio-title">Do you know how big is your species?</label>
      <div className="choices">
        <div className="firstCheck">
          <input type="radio" id="first" name="name" />
          <label for="first">Only few of us left...</label>
        </div>
        <div className="secondCheck">
          <input type="radio" id="second" name="name" />
          <label for="second">I think around 736,239</label>
        </div>
        <div className="thirdCheck">
          <input type="radio" id="third" name="name" />
          <label for="third">We are the whole continent!</label>
        </div>
        <div className="fourthCheck">
          <input type="radio" id="fourth" name="name" />
          <label for="fourth">Never been good at maths...</label>
        </div>
        <div className="fifthCheck">
          <input type="radio" id="fifth" name="name" />
          <label for="fifth">I don't want to disclose this</label>
        </div>
      </div>
    </div>
    <div className="checkbox-list">
      <label id="checkbox-title">Do you want extinction updates?</label>
      <div className="choices-checkbox">
        <div>
          <input type="checkbox" id="first-checkbox" value="checkbox-1" />

          <label for="first">
            Yes! If you can you deliver in my local river.
          </label>
        </div>
        <div>
          <input type="checkbox" id="first-checkbox" value="checkbox-1" />

          <label for="first">Yes please, love news and updates!</label>
        </div>
        <div>
          <input type="checkbox" id="first-checkbox" value="checkbox-1" />

          <label for="first">Maybe a good idea.</label>
        </div>
        <div>
          <input type="checkbox" id="first-checkbox" value="checkbox-1" />

          <label for="first">I will probably oversleep them anyway...</label>
        </div>
        <div>
          <input type="checkbox" id="first-checkbox" value="checkbox-1" />

          <label for="first">No! I don't care.</label>
        </div>
      </div>
    </div>
    <div className="comment">
      <label id="comment-title">If you can actually type, leave message</label>

      <textarea className="comment-text">
        here you can leave your comments and advices
      </textarea>
    </div>

    <button type="submit" id="submit">
      Send your Form
    </button>
    <button className="active-article__button">
      <Link to="/">Go Home</Link>
    </button>
  </section>
);

export default Survey;
