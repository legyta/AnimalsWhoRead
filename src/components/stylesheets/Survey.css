@import url("https://fonts.googleapis.com/css?family=Roboto:400,500&display=swap");
.survey--img {
  height: 20vh;
  width: 50%;
  margin-top: 7%;
  margin-bottom: 7%;
}
.survey-body {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  background-color: #101b5f;
  font-family: "Roboto";
}
.title-description .title-description {
  padding: 1rem 1rem;
}
.title-description .title h1 {
  font-size: 4vw;
  font-weight: 500;
  color: #f0a56b !important;
}
.title-description p {
  color: #e57a9b;
  font-weight: 500;
  font-size: 0.9rem;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}
@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .title-description .title h1 {
    font-size: 3rem;
  }
  .description p {
    font-size: 2rem;
  }
}
.contact--survey .name {
  display: flex;
  flex-direction: row;
  align-items: center;
}
.contact--survey .email {
  display: flex;
  flex-direction: row;
  align-items: center;
}
.contact--survey .age {
  display: flex;
  flex-direction: row;
  align-items: center;
}
.contact--survey #name {
  margin: 0 0.5rem;
  font-size: 0.8rem;
}
.contact--survey #email-input {
  margin: 0 0.74rem;
  font-size: 0.8rem;
}
.contact--survey #number {
  margin: 0 0.4rem;
  font-size: 0.8rem;
}
.contact--survey label {
  font-size: 0.9rem;
  color: #c6ecc7;
  margin: 0 1 0;
}
@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .contact label {
    font-size: 1.2rem;
  }
  .contact #name {
    font-size: 1rem;
  }
  .contact #email-input {
    font-size: 1rem;
  }
  .contact #number {
    font-size: 1rem;
    margin: 0 0.3rem;
  }
}
.dropdown-list {
  display: flex;
  flex-direction: column;
  width: 72%;
  margin: 1rem;
}
@media screen and (min-width: 684px) and (max-width: 767px) {
  .dropdown-list {
    width: 15%;
  }
}
@media screen and (min-width: 768px) and (max-width: 1024px) {
  .dropdown-list {
    width: 25%;
  }
}
@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .dropdown-list {
    width: 26%;
  }
}
.dropdown-list label {
  font-size: 1rem;
  color: #c6ecc7;
}
.dropdown-list #dropdown {
  font-size: 0.8rem;
}
@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .dropdown-list label {
    font-size: 1.2rem;
  }
  .dropdown-list #dropdown {
    font-size: 1rem;
  }
}
.radio-list #radio-title {
  margin: 2rem;
  color: #c6ecc7;
  font-size: 1rem;
}
.radio-list .choices {
  margin: 0 2rem;
  font-size: 0.8rem;
}
@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .radio-list #radio-title {
    font-size: 1.2rem;
  }
  .radio-list .choices {
    font-size: 1rem;
  }
}
.checkbox-list #checkbox-title {
  margin: 2rem;
  font-size: 1.2rem;
  color: #c6ecc7;
  font-size: 1rem;
}
.checkbox-list .choices-checkbox {
  margin: 0 2rem;
  font-size: 0.8rem;
}
@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .checkbox-list #checkbox-title {
    font-size: 1.2rem;
  }
  .checkbox-list .choices-checkbox {
    font-size: 1rem;
    margin: 0 1;
  }
}
.comment {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
#comment-title {
  font-size: 1rem;
  color: #c6ecc7;
  margin: 1.2rem;
}
.comment-text {
  width: 80%;
  font-size: 0.8rem;
}
@media screen and (min-width: 1025px) and (max-width: 1824px) {
  #comment-title {
    font-size: 1.2rem;
  }
  .comment-text {
    font-size: 1rem;
  }
}
button {
  color: #101b5f;
  background-color: #c6ecc7;
  margin: 1rem;
  font-size: 0.8rem;
}
@media screen and (min-width: 1025px) and (max-width: 1824px) {
  button {
    font-size: 1rem;
  }
}
#submit:hover {
  color: #101b5f;
  background-color: #f0a56b;
}
.active-article__button {
  margin: 0 0 1rem;
}
.dropdown-list #dropdown {
  font-size: 0.9rem;
  color: #e57a9b;
}
.dropdown-list label {
  font-size: 1.4rem;
  color: #c6ecc7;
  margin-top: 15%;
}
label {
  display: inline-block;
  color: #c6ecc7;
}
.choices {
  width: 50%;
}
