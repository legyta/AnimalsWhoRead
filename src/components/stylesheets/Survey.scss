/* COLORS */
$red: #f72143;
$intense-blue: #1f33f6;
$light-green: #c6ecc7;
$dark-blue: #101b5f;
$light-orange: #f0a56b;
$orange: #f85f28;
$green: #1e8059;
$pink: #e57a9b;
/* FONT FAMILY */
$font: "Roboto";
/* FONT WEIGHTS */
$light: 300;
$medium: 500;

@import url("https://fonts.googleapis.com/css?family=Roboto:400,500&display=swap");
.frame {
  height: 20%;
  width: 20%;
  margin: 2rem 2rem;
}

.survey-section {
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  background-color: $dark-blue;
  font-family: $font;
}

.survey-title-description {
  .survey-title-description {
    padding: 1rem 1rem;
  }

  .survey-title h1 {
    font-size: 1.7rem;
    font-weight: $medium;
    color: $light-orange;
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
  }

  .survey-p {
    color: $pink;
    font-weight: $medium;
    font-size: 0.9rem;
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
  }
}

// Font size on Laptops
@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .surevy-title-description .title h1 {
    font-size: 3rem;
  }

  .survey-description p {
    font-size: 2rem;
  }
}

.survey-contact {
  .survey-name {
    display: flex;
    flex-direction: row;
    align-items: center;
  }

  .email {
    display: flex;
    flex-direction: row;
    align-items: center;
  }

  .age {
    display: flex;
    flex-direction: row;
    align-items: center;
  }

  #name {
    margin: 0 0.5rem;
    font-size: 0.8rem;
  }

  #email-input {
    margin: 0 0.74rem;
    font-size: 0.8rem;
  }

  #number {
    margin: 0 0.4rem;
    font-size: 0.8rem;
  }

  label {
    font-size: 0.9rem;
    color: $light-green;
    margin: 0 1 0;
  }
}

// Font size on Laptops
@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .survey-contact label {
    font-size: 1.2rem;
  }

  .contact #name {
    font-size: 1rem;
  }

  .contact #email-input {
    font-size: 1rem;
  }

  .contact #number {
    font-size: 1rem;

    margin: 0 0.3rem;
  }
}

.dropdown-list {
  display: flex;
  flex-direction: column;
  width: 72%;
  margin: 1rem;
}

@media screen and (min-width: 768px) and (max-width: 1024px) {
  .dropdown-list {
    width: 35%;
  }
}

/* Logo on laptops (nav bar)*/

@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .dropdown-list {
    width: 26%;
  }
}

.dropdown-list {
  label {
    font-size: 1rem;
    color: $light-green;
  }

  #dropdown {
    font-size: 0.8rem;
  }
}

/* Logo on laptops (nav bar)*/

@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .dropdown-list {
    label {
      font-size: 1.2rem;
    }

    #dropdown {
      font-size: 1rem;
    }
  }
}

.radio-list {
  #radio-title {
    margin: 2rem;
    color: $light-green;
    font-size: 1rem;
  }

  .choices {
    margin: 0 2rem;
    font-size: 0.8rem;
  }

  input[type="radio"] {
    box-sizing: border-box;
    margin: 0.6rem;
  }
}

/* Logo on laptops (nav bar)*/

@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .radio-list {
    #radio-title {
      font-size: 1.2rem;
    }

    .choices {
      font-size: 1rem;
    }
  }
}

.checkbox-list {
  #checkbox-title {
    margin: 2rem;
    font-size: 1.2rem;
    color: $light-green;
    font-size: 1rem;
  }

  .choices-checkbox {
    margin: 0 2rem;
    font-size: 0.8rem;
  }

  input[type="checkbox"],
  input[type="radio"] {
    box-sizing: border-box;
    margin: 0.6rem;
  }
}

@media screen and (min-width: 1025px) and (max-width: 1824px) {
  .checkbox-list {
    #checkbox-title {
      font-size: 1.2rem;
    }

    .choices-checkbox {
      font-size: 1rem;
    }
  }
}

.comment {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

#comment-title {
  font-size: 1rem;
  color: $light-green;
  margin: 1.2rem;
}

.comment-text {
  width: 80%;
  font-size: 0.8rem;
}

@media screen and (min-width: 1025px) and (max-width: 1824px) {
  #comment-title {
    font-size: 1.2rem;
  }

  .comment-text {
    font-size: 1rem;
  }
}

button {
  color: $dark-blue;
  background-color: $light-green;
  margin: 1rem;
  font-size: 0.8rem;
}

@media screen and (min-width: 1025px) and (max-width: 1824px) {
  button {
    font-size: 1rem;
  }
}

#submit:hover {
  color: $dark-blue;
  background-color: $light-orange;
}

.active-article__button {
  margin: 0 0 1rem;
}
