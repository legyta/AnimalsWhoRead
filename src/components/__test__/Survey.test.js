import React from "react";
import Survey from "../Survey";
import { Item } from "react-bootstrap/lib/Pagination";

test("shallow rendering", () => {
  const wrapper = shallow(<Survey />);
  expect(wrapper).toMatchSnapshot();
});
