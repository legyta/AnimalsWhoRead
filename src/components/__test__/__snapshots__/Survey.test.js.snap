// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow rendering 1`] = `
<div
  className="survey-section"
  id="survey"
>
  <NavBar />
  <SideDrawer />
  <img
    alt="frame"
    className="frame"
    src="frame.svg"
  />
  <div
    className="survey-title-description"
  >
    <div
      className="survey-title"
    >
      <h1>
        Feelings about Extinction
      </h1>
    </div>
    <div
      className="description"
    >
      <p
        className="survey-p"
      >
        Greetings to animals sharing their feelings here
      </p>
    </div>
  </div>
  <div
    className="survey-contact"
  >
    <div
      className="survey-name"
    >
      <label
        htmlFor="text"
        id="name-label"
      >
        How are you called
      </label>
      <input
        className="surveyContact-input"
        id="survey-name"
        name="name"
        placeholder="your species name"
        required={true}
        type="survey-text"
      />
    </div>
    <div
      className="survey-email"
    >
      <label
        htmlFor="email"
        id="email-label"
      >
        What is your email
      </label>
      <input
        className="surveyContact-input"
        id="email-input"
        name="email"
        pattern="[^ @]*@[^ @]*"
        placeholder="if you have one"
        type="survey-email"
      />
    </div>
    <div
      className="survey-age"
    >
      <label
        htmlFor="number"
        id="number-label"
      >
        Years on the planet
      </label>
      <input
        className="surveyContact-input"
        id="number"
        max="100"
        min="0"
        name="number"
        placeholder="age"
        type="survey-number"
      />
    </div>
  </div>
  <div
    className="dropdown-list"
  >
    <label
      className="dropdown-title"
    >
      How do you feel about Extinction?
    </label>
    <select
      className="dropdown"
    >
      <option
        value="current"
      >
        Choose what you relate most to
      </option>
      <option
        value="first-choice"
      >
        I am very afraid! I already store food and think of new possible shelter.
      </option>
      <option
        value="second-choice"
      >
        We have a plan to fight it!
      </option>
      <option
        value="third-choice"
      >
        I am scared, but I don't care. My elders also got extincted.
      </option>
      <option
        value="fourth-choice"
      >
        Huh? What is that? Did I sleep for too long again. Need to catch up on the news!
      </option>
      <option
        value="fifth-choice"
      >
        Prefer not to answer.
      </option>
    </select>
  </div>
  <div
    className="radio-list"
  >
    <label
      className="radio-title"
    >
      Do you know how big is your species?
    </label>
    <div
      className="choices"
    >
      <div
        className="firstCheck"
      >
        <input
          id="first"
          name="name"
          type="radio"
        />
        <label
          htmlFor="first"
        >
          Only few of us left...
        </label>
      </div>
      <div
        className="secondCheck"
      >
        <input
          id="second"
          name="name"
          type="radio"
        />
        <label
          htmlFor="second"
        >
          I think around 736,239
        </label>
      </div>
      <div
        className="thirdCheck"
      >
        <input
          id="third"
          name="name"
          type="radio"
        />
        <label
          htmlFor="third"
        >
          We are the whole continent!
        </label>
      </div>
      <div
        className="fourthCheck"
      >
        <input
          id="fourth"
          name="name"
          type="radio"
        />
        <label
          htmlFor="fourth"
        >
          Never been good at maths...
        </label>
      </div>
      <div
        className="fifthCheck"
      >
        <input
          id="fifth"
          name="name"
          type="radio"
        />
        <label
          htmlFor="fifth"
        >
          I don't want to disclose this
        </label>
      </div>
    </div>
  </div>
  <div
    className="checkbox-list"
  >
    <label
      className="checkbox-title"
    >
      Do you want extinction updates?
    </label>
    <div
      className="choices-checkbox"
    >
      <div
        className="firstTick"
      >
        <input
          className="check"
          id="first-checkbox"
          type="checkbox"
          value="checkbox-1"
        />
        <label
          htmlFor="first"
        >
          If you can you deliver to my local river!
        </label>
      </div>
      <div>
        <input
          className="check"
          id="second-checkbox"
          type="checkbox"
          value="checkbox-1"
        />
        <label
          htmlFor="second"
        >
          Yes please, love news and updates!
        </label>
      </div>
      <div>
        <input
          className="check"
          id="third-checkbox"
          type="checkbox"
          value="checkbox-1"
        />
        <label
          htmlFor="third"
        >
          Maybe a good idea.
        </label>
      </div>
      <div>
        <input
          className="check"
          id="fourth-checkbox"
          type="checkbox"
          value="checkbox-1"
        />
        <label
          htmlFor="fourth"
        >
          I will probably oversleep them anyway...
        </label>
      </div>
      <div>
        <input
          className="check"
          id="fifth-checkbox"
          type="checkbox"
          value="checkbox-1"
        />
        <label
          htmlFor="fifth"
        >
          No! I don't care.
        </label>
      </div>
    </div>
  </div>
  <div
    className="comment"
  >
    <label
      className="comment-title"
    >
      If you can actually type, leave message
    </label>
    <textarea
      className="comment-text"
      placeholder="here you can leave your comments and advices"
    />
  </div>
  <button
    className="survey-button"
    id="submit"
    onClick={[Function]}
    type="submit"
  >
    Send your Form
  </button>
  <button
    className="active-article__button"
  >
    <Link
      to="/"
    >
      Go Home
    </Link>
  </button>
  <footer
    className="survey-footer"
  >
    <p>
      We will fight 
      <br />
      back.
    </p>
    <span>
      Â© 2020 
      <br />
      by Aigul, Birte, Ligita,
      <br />
      Seyma
      <br />
      <br />
      #wildcodeschool
    </span>
  </footer>
</div>
`;
